seps/sep-0002.md

```diff
@@ -0,0 +1,225 @@
+
+3. With the above we can agree to use the namespace @ `grupos.near/post/index` to contain groups:
+
+{
+    "alice.near": {...},
+    "mob.near": {...},
+    "grupos.near":
+    {
+        "post":
+        {
+            "index":
+            {
+                "1290387aslkdjf09182_members":
+                {
+                    "alice.near": "",
+					"bob.near": "",
+                },
+                "1290387aslkdjf09182_data":
+                {
+                    "name": "zksync fans",
+                    "gateway": "near.org",
+                    "metadata":
+                    {
+                        "description": "we love zksync",
+                        "tags": ["ether-js", "eth", "multichain"]
+                    }
+                },
+                "00110357asiedjf0935t_members":
+                {
+                    "gauthamravi.near": "",
+                    "mob.near": "",
+                    "root.near": ""
+                },
+                 "00110357asiedjf0935t_data":
+                {
+                    "name": "the best group",
+                    "gateway": "near.social",
+                    "metadata":
+                    {
+                        "description": "super group",
+                        "guidelines": "be excellent humans."
+                    }
+                }
+            }
+        }
+    }
+}
+
+### Supporting this requires: 
+
+1. A common goods provider to maintain a service that abstracts and securely exposes calls to social.near's `grant_write_access` and `set` methods via an API such as
+	/groups/[groupUID]/add/[accountId]  (adds a user to a group and creates the group if it doesn't exist)
+	/groups/[groupUID]/remove/[accountId] 
+	/groups/[groupUID]/verify?accountIds=[accountIds] (a convenience method to expose the canonical source of truth of group membership. Returns an object of [accountId]:boolean pairs to confirm which accountIds have write access fo the `_members` scope of the given groupUID. This API essentially wraps calls to `is_write_permission_granted`, modified to accept multiple AccountIds.)
+2. A common goods provider to maintain an indexer and service to fan-out group notifications into notifications sent to each group's member.
+
+
+
+## Drawbacks
+
+- By granting each group member write access to `grupos.near/post/index/[groupUID]_members`, any group member could leverage near.social's `set` to remove others from the list of group members. However, this is of minimal impact since this list is independent of the management of write permissions. In this case, this means that a removed group member would retain their ablity to write the list of members. Easily supporting data recovery from a previous block.  
```

Comment:
How would this effect scenarios where group members actually would need to be removed due to behavior that goes against a group's guidelines? Would they need to be removed from both the member list and write permissions list?

Reply:
That's correct. A gateway operator would need to couple the mutation of the write permissions and the member AccountList on the UI affordance which adds/removes group members.